aea/protocols/generator.py

```diff
@@ -922,6 +930,116 @@ def _encoding_message_field_from_python_to_protobuf(
             )
         return encoding_str
 
+    def _decoding_message_field_from_protobuf_to_python(
+        self,
+        performative,
+        content_name,
+        content_type,
+        no_indents,
+        variable_name_in_protobuf: Optional[str] = "",
+    ) -> str:
+        """
+        Produce the decoding of message contents for the serialisation class.
+
+        :param performative: the performative to which the content belongs
+        :param content_name: the name of the content to be decoded
+        :param content_type: the type of the content to be decoded
+        :param no_indents: the number of indents based on the previous sections of the code
+        :return: the decoding string
+        """
+        decoding_str = ""
+        indents = get_indent_str(no_indents)
+        if content_type in PYTHON_TYPE_TO_PROTO_TYPE.keys():
+            if variable_name_in_protobuf == "":
+                decoding_str += indents + "{} = {}_pb.{}.{}\n".format(
+                    content_name,
+                    self.protocol_specification.name,
+                    performative,
+                    content_name,
+                )
+            else:
+                decoding_str += indents + "{} = {}_pb.{}.{}\n".format(
+                    content_name,
+                    self.protocol_specification.name,
+                    performative,
+                    variable_name_in_protobuf,
+                )
+            decoding_str += indents + 'performative_content["{}"] = {}\n'.format(
+                content_name, content_name
+            )
+        elif content_type.startswith("FrozenSet") or content_type.startswith("Tuple"):
+            if variable_name_in_protobuf == "":
```

Comment:
fine like it is, but it would be slightly less verbose if you did introduce a helper method:
variable_name = content_name if variable_name_in_protobuf == "" else variable_name_in_protobuf

Reply:
Done